datatable_tutorial/datatable_tutorial/datatable_tutorial.py

```diff
@@ -0,0 +1,269 @@
+        self.right_clicked_group_header = f"Group header right clicked at index: {index}, Group header value: {val['group']}"
+
+    def get_item_hovered(self, pos) -> str:
+        self.item_hovered = f"Item Hovered type: {pos['kind']}, Location: {pos['location']}"
+        
+    def get_deleted_item(self, selection):
+        self.deleted = f"Deleted cell: {selection['current']['cell']}"
+
+    # def append_row(self):
+    #     print("13232")
+
+    def column_resize(self, col, width):
+        self.cols[col['pos']]['width'] = width 
+
+
+
+
+class DataTableLiveState(BaseState):
+    "The app state."
+
+    running: bool
+    table_data: list[dict[str, Any]] = []
+    rate: int = 0.4
+    columns: list[dict[str, str]] = [
+        {
+            "title": "id", 
+            "id": "v1", 
+            "type": "int",
+            "width": 100,
+        },
+        {
+            "title": "advice", 
+            "id": "v2", 
+            "type": "str",
+            "width": 750,
+        },
+    
+    ]
+
+    @rx.background
+    async def live_stream(self):
+        while True:
+            await asyncio.sleep(1 / self.rate)
+            if not self.running:
+                break
+
+            async with self:
+                if len(self.table_data) > 50:
+                    self.table_data.pop(0)
+
+                res = httpx.get('https://api.adviceslip.com/advice')
+                data = res.json()
+                self.table_data.append({"v1": data["slip"]["id"], "v2": data["slip"]["advice"]})
+
+
+    def toggle_pause(self):
+        self.running = not self.running
+        if self.running:
+            return DataTableLiveState.live_stream
+
+
+darkTheme = {
+    "accentColor": "#8c96ff",
+    "accentLight": "rgba(202, 206, 255, 0.253)",
+    "textDark": "#ffffff",
+    "textMedium": "#b8b8b8",
+    "textLight": "#a0a0a0",
+    "textBubble": "#ffffff",
+    "bgIconHeader": "#b8b8b8",
+    "fgIconHeader": "#000000",
+    "textHeader": "#a1a1a1",
+    "textHeaderSelected": "#000000",
+    "bgCell": "#16161b",
+    "bgCellMedium": "#202027",
+    "bgHeader": "#212121",
+    "bgHeaderHasFocus": "#474747",
+    "bgHeaderHovered": "#404040",
+    "bgBubble": "#212121",
+    "bgBubbleSelected": "#000000",
+    "bgSearchResult": "#423c24",
+    "borderColor": "rgba(225,225,225,0.2)",
+    "drilldownBorder": "rgba(225,225,225,0.4)",
+    "linkColor": "#4F5DFF",
+    "headerFontStyle": "bold 14px",
+    "baseFontStyle": "13px",
+    "fontFamily": "Inter, Roboto, -apple-system, BlinkMacSystemFont, avenir next, avenir, segoe ui, helvetica neue, helvetica, Ubuntu, noto, arial, sans-serif",
+}
+
+def index() -> rx.Component:
+    return rx.fragment(
+        rx.color_mode_button(rx.color_mode_icon(), float="right"),
+        rx.vstack(
+            rx.heading("Data Table Demo!", font_size="2em"),
+            rx.vstack(
+                rx.button("Static / Live Data", on_click=DataTableState.change, color_scheme='messenger', size='lg',),
```

Comment:
i think this would be better as a tab component to switch between static or live. as a button, it's not obvious what it's doing

Reply:
switched it to tabs